Tasks/UseGoV1/installer.ts

```diff
@@ -0,0 +1,118 @@
+import * as toolLib from 'azure-pipelines-tool-lib/tool';
+import * as tl from 'azure-pipelines-task-lib/task';
+
+import * as path from 'path';
+import * as util from 'util';
+import * as os from 'os';
+
+let osPlat: string = os.platform();
+let osArch: string = os.arch();
+
+export async function getGo(version: string) {
+    // check cache
+    let toolPath: string;
+    toolPath = toolLib.findLocalTool('go', fixVersion(version));
+
+    if (!toolPath) {
+        // download, extract, cache
+        toolPath = await acquireGo(version);
+        tl.debug("Go tool is cached under " + toolPath);
+    }
+
+    setGoEnvironmentVariables(toolPath);
+
+    toolPath = path.join(toolPath, 'bin');
+    //
+    // prepend the tools path. instructs the agent to prepend for future tasks
+    //
+    toolLib.prependPath(toolPath);
+}
+
+
+async function acquireGo(version: string): Promise<string> {
+    //
+    // Download - a tool installer intimately knows how to get the tool (and construct urls)
+    //
+    let fileName: string = getFileName(version);
+    let downloadUrl: string = getDownloadUrl(fileName);
+    let downloadPath: string = null;
+    try {
+        downloadPath = await toolLib.downloadTool(downloadUrl);
+    } catch (error) {
+        tl.debug(error);
+
+        // cannot localized the string here because to localize we need to set the resource file.
+        // which can be set only once. vsts-task-tool-lib/tool, is already setting it to different file.
+        // So left with no option but to hardcode the string. Other tasks are doing the same.
+        throw (util.format("Failed to download version %s. Please verify that the version is valid and resolve any other issues. %s", version, error));
```

Comment:
"And resolve any other issues." Maybe drop this? Not helpful.

Should we add the error in the thrown message to make it easier to debug?

Reply:
> Should we add the error in the thrown message to make it easier to debug?

Aren't we already doing this?

Reply:
Dropped the resolve other issues